### DISCUSSION TITLE
Intent Recognition not very good?

### DISCUSSION THREAD

**Discussion by KevinAHM** [deepgram/community#696] [General help]
Am I doing something wrong? The intent API doesn't seem production ready. No matter what I set in the intent or topic boxes it detects nothing. I've tried many things including what's in this screenshot:
![Screenshot 2024-04-11 213322](https://github.com/deepgram/community/assets/131084116/d3d48745-4a3b-473f-8c85-0c6b30b519de)
![Screenshot 2024-04-11 213327](https://github.com/deepgram/community/assets/131084116/e975859e-f347-4ce3-8176-1f52f3f6f072)

Meanwhile I can get lower latency and better performance out of a GPT 3.5 prompt I made in 5 minutes.

> # Accept Input Text:
> ## Receive a text string from the user. This text could be one or more sentences long.
> ## Segmentation: Divide the input text into logical segments. These segments could be based on any of the following:
> 
> - Sentences
> - Changes in topic or idea
> - Grammatical structures (like dependent clauses)
> 
> # Intent Classification: For each segment, perform the following:
> 
> ## Identify the most likely intent behind the text segment. Possible intents could include (but are not limited to):
> - Question
> - Statement of fact
> - Request
> - Command
> - Expression of opinion
> ## Additional intents:
> - Product inquiry
> ## Assign a confidence score between 0.000 and 1.000 to the identified intent, reflecting your certainty about the classification.
> 
> # Finally:
> All messages sent to you are meant to be intent classified following the above criteria.

![Screenshot 2024-04-11 212805](https://github.com/deepgram/community/assets/131084116/2775026d-cdb7-4070-8c42-8b70ac2dad0c)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @kjjk10, thanks for sharing this issue. I think there are a few factors.

One is that our models were trained on generally longer input in a conversational context, so one short sentence is a less common input. 

Another factor with the custom intents is that they need to be in verb form. We have one small doc note about this on the [Intent Recognition](https://developers.deepgram.com/docs/intent-recognition) page:
> `intent`: The name of the intent detected by the model. This will always be in the form of a verb.

For instance, your intent of "product inquiry" would need to be "inquire about a product"; "purchasing" would be "purchase", etc. "Buy" and "buy art" would be valid as-is.

Finally, I see one issue that adding a period to input a complete sentence does yield the intent "buy bronze statue". 

![Screenshot 2024-04-12 at 2 27 40 PM](https://github.com/deepgram/community/assets/135150417/910f90d4-82c9-40bd-b1bb-b18aa370d9b7)

When I turn on "restrict to only custom intents," I still do not get any intents from your example. 

I hope this information helps you continue investigating Deepgram, and we'll take your experience as feedback back into the feature.

**Reply by KevinAHM** [reply]
It seems there's significant room for improvement. For a classifier to be production-ready, it needs to handle basic sentence structures and punctuation variations reliably.

![Screenshot 2024-04-12 222810](https://github.com/deepgram/community/assets/131084116/0cf30010-5578-4336-9422-41276c3b2357)

**Comment by KevinAHM** [comment]
> Hi @kjjk10, thanks for sharing this issue. I think there are a few factors.
> 
> One is that our models were trained on generally longer input in a conversational context, so one short sentence is a less common input.
> 
> Another factor with the custom intents is that they need to be in verb form. We have one small doc note about this on the [Intent Recognition](https://developers.deepgram.com/docs/intent-recognition) page:
> 
> > `intent`: The name of the intent detected by the model. This will always be in the form of a verb.
> 
> For instance, your intent of "product inquiry" would need to be "inquire about a product"; "purchasing" would be "purchase", etc. "Buy" and "buy art" would be valid as-is.
> 
> Finally, I see one issue that adding a period to input a complete sentence does yield the intent "buy bronze statue".
> 
> ![Screenshot 2024-04-12 at 2 27 40 PM](https://private-user-images.githubusercontent.com/135150417/322116082-910f90d4-82c9-40bd-b1bb-b18aa370d9b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5ODYwOTMsIm5iZiI6MTcxMjk4NTc5MywicGF0aCI6Ii8xMzUxNTA0MTcvMzIyMTE2MDgyLTkxMGY5MGQ0LTgyYzktNDBiZC1iMWJiLWIxOGFhMzcwZDliNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxM1QwNTIzMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MjNkMTJhNDFjNWFiMjg5YjNhZjc2ZWRhMDc1MWY0OTE4MGE1NTA0N2Y5ZTEyMGFmYTg0YzgyZmU5ZDhmYTIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.n1J3mRsTwOZdPwuGlB3RV3Pi7tLFunvoMPH64kcOxjs)
> 
> When I turn on "restrict to only custom intents," I still do not get any intents from your example.
> 
> I hope this information helps you continue investigating Deepgram, and we'll take your experience as feedback back into the feature.
